cmd/kube-controller-manager/app/cloudproviders.go

```diff
@@ -32,29 +30,28 @@ import (
 func createCloudProvider(logger klog.Logger, cloudProvider string, externalCloudVolumePlugin string, cloudConfigFile string,
```

Comment:
i tend to agree that we can clean up these functions but i wanted to make smaller changes because i think there are some functions that depend on the default values returning. i noticed that when i tried to remove more of the functionality around the default loops that there were some crashes that need investigating.